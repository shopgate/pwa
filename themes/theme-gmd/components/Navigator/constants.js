/**
  * Action Types.
  */

export const TOGGLE_NAVIGATOR_CART_ICON = 'TOGGLE_NAVIGATOR_CART_ICON';
export const TOGGLE_NAVIGATOR_SEARCH = 'TOGGLE_NAVIGATOR_SEARCH';
export const TOGGLE_NAV_DRAWER = 'TOGGLE_NAV_DRAWER';
export const TOGGLE_PROGRESS_BAR = 'TOGGLE_PROGRESS_BAR';
export const TOGGLE_LOGIN = 'TOGGLE_LOGIN';
export const SET_NAVIGATOR_BACKGROUND = 'SET_NAVIGATOR_BACKGROUND';
export const SET_NAVIGATOR_COLOR = 'SET_NAVIGATOR_COLOR';
export const SET_NAVIGATOR_SEARCH_QUERY = 'SET_NAVIGATOR_SEARCH_QUERY';
export const SET_FILTER_OPENED = 'SET_FILTER_OPENED';
export const SET_FILTER_CLOSED = 'SET_FILTER_CLOSED';
export const SET_FILTER_ATTRIBUTE_OPENED = 'SET_FILTER_ATTRIBUTE_OPENED';
export const SET_FILTER_ATTRIBUTE_CLOSED = 'SET_FILTER_ATTRIBUTE_CLOSED';
export const SET_SEARCH_ENABLED = 'SET_SEARCH_ENABLED';
export const SET_SEARCH_DISABLED = 'SET_SEARCH_DISABLED';
export const SET_ICON_SHADOW_ENABLED = 'SET_ICON_SHADOW_ENABLED';
export const SET_ICON_SHADOW_DISABLED = 'SET_ICON_SHADOW_DISABLED';
export const SET_TITLE_ENABLED = 'SET_TITLE_ENABLED';
export const SET_TITLE_DISABLED = 'SET_TITLE_DISABLED';
export const SET_NAVIGATOR_DISABLED = 'SET_NAVIGATOR_DISABLED';
export const SET_NAVIGATOR_ENABLED = 'SET_NAVIGATOR_ENABLED';
