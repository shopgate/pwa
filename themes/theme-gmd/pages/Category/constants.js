export const GRID_VIEW = 'grid';
export const LIST_VIEW = 'list';

export const SET_CATEGORY_VIEW_MODE = 'SET_CATEGORY_VIEW_MODE';
