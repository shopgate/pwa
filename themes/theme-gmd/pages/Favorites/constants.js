export const FAVORITES_SHOW_TOAST_DELAY = 300;
export const FAVORITES_SHOW_LIMIT = 10;
