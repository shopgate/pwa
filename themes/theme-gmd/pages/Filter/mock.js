const mockedState = {
  filter: {
    activeFilters: [
      {
        filters: {},
        categoryId: 'men',
      },
    ],
    availableFilters: {
      '{"categoryId":"men","filters":{},"pipeline":"shopgate.catalog.getFilters"}': {
        filters: [
          {
            id: 'display_amount',
            label: 'Price',
            type: 'range',
            minimum: 1250,
            maximum: 3499,
            url: '/filter/display-amount',
          },
          {
            id: 'manufacturer',
            label: 'Brand',
            type: 'multiselect',
            source: 'manufacturer',
            values: [
              {
                id: 'Acme',
                label: 'Acme',
                hits: 1,
              },
              {
                id: 'Material Fashion',
                label: 'Material Fashion',
                hits: 3,
              },
            ],
            url: '/filter/manufacturer',
          },
          {
            id: 'Color',
            label: 'Color',
            type: 'multiselect',
            source: 'attributes',
            values: [
              {
                id: 'Blue',
                label: 'Blue',
                hits: 1,
              },
              {
                id: 'Green',
                label: 'Green',
                hits: 1,
              },
              {
                id: 'Red',
                label: 'Red',
                hits: 1,
              },
            ],
            url: '/filter/color',
          },
          {
            id: 'Length',
            label: 'Length',
            type: 'multiselect',
            source: 'attributes',
            values: [
              {
                id: '60',
                label: '60',
                hits: 1,
              },
              {
                id: '80',
                label: '80',
                hits: 1,
              },
            ],
            url: '/filter/length',
          },
          {
            id: 'Size',
            label: 'Size',
            type: 'multiselect',
            source: 'attributes',
            values: [
              {
                id: '2X-Large',
                label: '2X-Large',
                hits: 1,
              },
              {
                id: '3X-Large',
                label: '3X-Large',
                hits: 1,
              },
              {
                id: 'Large',
                label: 'Large',
                hits: 1,
              },
              {
                id: 'Medium',
                label: 'Medium',
                hits: 1,
              },
              {
                id: 'Small',
                label: 'Small',
                hits: 1,
              },
              {
                id: 'X-Large',
                label: 'X-Large',
                hits: 1,
              },
              {
                id: 'X-Small',
                label: 'X-Small',
                hits: 1,
              },
            ],
            url: '/filter/size',
          },
          {
            id: 'Title',
            label: 'Title',
            type: 'multiselect',
            source: 'attributes',
            values: [
              {
                id: 'Lithograph - Height: 9" x Width: 12"',
                label: 'Lithograph - Height: 9" x Width: 12"',
                hits: 1,
              },
              {
                id: 'Medium',
                label: 'Medium',
                hits: 1,
              },
              {
                id: 'Small',
                label: 'Small',
                hits: 1,
              },
            ],
            url: '/filter/title',
          },
          {
            id: 'Width',
            label: 'Width',
            type: 'multiselect',
            source: 'attributes',
            values: [
              {
                id: '20',
                label: '20',
                hits: 1,
              },
              {
                id: '40',
                label: '40',
                hits: 1,
              },
            ],
            url: '/filter/width',
          },
          {
            id: 'Category',
            label: 'Category',
            type: 'single_select',
            source: 'factfinder',
            values: [
              {
                id: '20',
                label: '20',
                hits: 1,
              },
              {
                id: '40',
                label: '40',
                hits: 1,
              },
            ],
            url: '/filter/category',
          },
        ],
        isFetching: false,
        expires: 1524061454824,
      },
    },
    activeHash: '{"categoryId":"men","filters":{},"pipeline":"shopgate.catalog.getFilters"}',
    temporaryFilters: {},
  },
  history: {
    action: 'PUSH',
    pathname: '/filter',
    queryParams: {},
    length: 2,
    redirectLocation: null,
  },
};

const mockedEmpty = {
  ...mockedState,
  ...{
    filter: {
      activeFilters: [],
      availableFilters: {},
      activeHash: '',
      temporaryFilters: {},
    },
  },
};

export {
  mockedState,
  mockedEmpty,
};
