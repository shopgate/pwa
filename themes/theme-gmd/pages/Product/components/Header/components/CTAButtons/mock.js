export const mockedState = {
  product: {
    currentProduct: {
      productId: 'product_1',
    },
    productsById: {
      product_1: {
        isFetching: false,
        productData: {
          id: 'product_1',
          name: 'Product No. 1',
          flags: {
            hasVariants: true,
            hasOptions: false,
          },
        },
      },
      product_2: {
        isFetching: false,
        productData: {
          id: 'product_2',
          baseProductId: 'product_1',
          name: 'Product No. 2',
          flags: {
            hasVariants: false,
            hasOptions: false,
          },
        },
      },
    },
  },
  favorites: {
    products: {
      isFetching: false,
      ids: ['product_1', 'product_2'],
    },
  },
  cart: {
    productPendingCount: 0,
  },
};

export const mockedVariantState = {
  product: {
    currentProduct: {
      productId: 'product_1',
      productVariantId: 'product_2',
    },
    productsById: {
      product_1: {
        isFetching: false,
        productData: {
          id: 'product_1',
          name: 'Product No. 1',
          flags: {
            hasVariants: true,
            hasOptions: false,
          },
        },
      },
      product_2: {
        isFetching: false,
        productData: {
          id: 'product_2',
          baseProductId: 'product_1',
          name: 'Product No. 2',
          flags: {
            hasVariants: false,
            hasOptions: false,
          },
        },
      },
    },
    optionsByProductId: {
      product_1: null,
      product_2: null,
    },
  },
  favorites: {
    products: {
      isFetching: false,
      ids: ['product_1', 'product_2'],
    },
  },
  cart: {
    productPendingCount: 0,
  },
};
