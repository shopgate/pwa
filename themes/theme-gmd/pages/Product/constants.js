/**
 * Action Types
 */
export const OPEN_PRODUCT_GALLERY = 'OPEN_PRODUCT_GALLERY';
export const REQUEST_PRODUCT_DATA = 'REQUEST_PRODUCT_DATA';
