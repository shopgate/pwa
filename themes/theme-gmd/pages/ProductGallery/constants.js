/**
 * key in the product image collection
 */
export const GALLERY_SLIDER_IMAGE_COLLECTION_KEY = 'GALLERY_SLIDER_IMAGES';

/**
 * themes value for the product slider image formats
 */
export const GALLERY_SLIDER_IMAGE_FORMATS = [
  {
    width: 1024,
    height: 1024,
  },
  {
    width: 2048,
    height: 2048,
  },
];

export const GALLERY_SLIDER_ZOOM = {
  enabled: true,
  maxRatio: 4,
  minRation: 1,
};

