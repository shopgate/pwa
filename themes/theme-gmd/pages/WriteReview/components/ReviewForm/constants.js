export const DEFAULT_FORM_MAX_LENGTH = (2 ** 8) - 1;
export const REVIEW_TEXT_MAX_LENGTH = (2 ** 15) - 1;

export const FIELD_NAME_AUTHOR = 'author';
export const FIELD_NAME_TITLE = 'title';
export const FIELD_NAME_REVIEW = 'review';
export const FIELD_NAME_RATE = 'rate';
