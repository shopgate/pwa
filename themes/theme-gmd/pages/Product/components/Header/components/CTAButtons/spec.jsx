import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import { ADD_PRODUCTS_TO_CART } from '@shopgate/pwa-common-commerce/cart/constants';
import { MockedAddToCartButton } from '@shopgate/pwa-ui-shared/AddToCartButton/mock';
import {
  mockedState,
  mockedVariantState,
} from './mock';

const mockedButton = MockedAddToCartButton;
jest.mock('@shopgate/pwa-ui-shared/AddToCartButton', () => mockedButton);

/**
 * Mock add to cart action
 * @return {{type}}
 */
const mockAddProductsToCart = () => ({
  type: ADD_PRODUCTS_TO_CART,
});

jest.mock('@shopgate/pwa-common-commerce/cart/actions/addProductsToCart.js', () => mockAddProductsToCart);

describe('CTAs (product header)', () => {
  let store;
  /**
   * Creates component
   * @param {Object} state Mocked state
   * @return {ReactWrapper}
   */
  const createComponent = (state) => {
    const mockedStore = configureStore([thunk]);
    store = mockedStore(state);

    /* eslint-disable global-require */
    const CTAButtons = require('./index').default;
    /* eslint-enable global-require */
    return mount(<Provider store={store}><CTAButtons /></Provider>);
  };

  describe('Rendering', () => {
    it('should render CTAButtons when data is available', () => {
      const component = createComponent(mockedState);
      expect(component).toMatchSnapshot();
      expect(component.find('FavoritesButton').exists()).toBe(true);
      expect(component.find('MockedAddToCartButton').exists()).toBe(true);
    });

    it('should handle add to cart', () => {
      const component = createComponent(mockedVariantState);
      component.find('MockedAddToCartButton').simulate('click');
      const actions = store.getActions();
      expect(actions[0].type).toBe(ADD_PRODUCTS_TO_CART);
    });
  });
});
