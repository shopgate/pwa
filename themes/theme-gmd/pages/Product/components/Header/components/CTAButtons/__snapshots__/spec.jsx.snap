// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAs (product header) Rendering should render CTAButtons when data is available 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CTAButtons)>
    <CTAButtons
      handleAddToCart={[Function]}
      isDisabled={false}
      isFavorite={true}
      isLoading={true}
      isOrderable={false}
      productId="product_1"
    >
      <div
        className="css-7azkrx"
      >
        <Connect(FavoritesButton)
          active={true}
          className="css-13l587d"
          productId="product_1"
          rippleClassName="css-1tdptiz"
        >
          <FavoritesButton
            active={true}
            addFavorites={[Function]}
            className="css-13l587d"
            noShadow={false}
            onRippleComplete={[Function]}
            once={false}
            productId="product_1"
            removeFavorites={[Function]}
            removeThrottle={0}
            removeWithRelatives={false}
            rippleClassName="css-1tdptiz"
          />
        </Connect(FavoritesButton)>
        <MockedAddToCartButton
          buttonSize={56}
          handleAddToCart={[Function]}
          iconSize={24}
          isDisabled={false}
          isLoading={true}
          isOrderable={false}
        >
          <button
            onClick={[Function]}
          />
        </MockedAddToCartButton>
      </div>
    </CTAButtons>
  </Connect(CTAButtons)>
</Provider>
`;
